acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/acceptance/dsl/node/cluster/Cluster.java

```diff
@@ -88,33 +89,85 @@ public void start(final List<RunnableNode> nodes) {
   }
 
   public void stop() {
-    for (final RunnableNode node : nodes.values()) {
-      node.stop();
+    for (final NodeInfo nodeInfo : nodes.values()) {
+      nodeInfo.getRunnableNode().stop();
+      nodeInfo.setRunning(false);
     }
     pantheonNodeRunner.shutdown();
   }
 
   public void stopNode(final PantheonNode node) {
+    nodes.get(node.getName()).setRunning(false);
     pantheonNodeRunner.stopNode(node);
   }
 
   @Override
   public void close() {
-    for (final RunnableNode node : nodes.values()) {
-      node.close();
+    for (final NodeInfo nodeInfo : nodes.values()) {
+      nodeInfo.getRunnableNode().close();
+      nodeInfo.setRunning(false);
     }
     pantheonNodeRunner.shutdown();
   }
 
   public void verify(final Condition expected) {
-    for (final Node node : nodes.values()) {
+    for (final NodeInfo nodeInfo : nodes.values()) {
+      expected.verify(nodeInfo.getRunnableNode());
+    }
+  }
+
+  public void verifyRunningNodes(final Condition expected) {
+    getRunningNodes().forEach(expected::verify);
+  }
+
+  public void verify(final Condition expected, final Collection<Node> nodes) {
+    for (final Node node : nodes) {
       expected.verify(node);
     }
   }
 
   public void waitUntil(final WaitCondition condition) {
-    for (final Node node : nodes.values()) {
-      node.waitUntil(condition);
+    for (final NodeInfo nodeInfo : nodes.values()) {
+      nodeInfo.getRunnableNode().waitUntil(condition);
+    }
+  }
+
+  public List<RunnableNode> getRunningNodes() {
+    return nodes
+        .values()
+        .stream()
+        .filter(nodeInfo -> nodeInfo.isRunning)
+        .map(NodeInfo::getRunnableNode)
+        .collect(toList());
+  }
+
+  public RunnableNode getFirstRunningNode() {
+    return getRunningNodes().get(0);
+  }
+
+  private static class NodeInfo {
```

Comment:
What's the value in this new inner class, my IntelliJ shows `isRunning()` and `isRunning` are not called outside NodeInfo.

Reply:
It's just way a way adding of associating a the running state with the nodes and their names. Allows me to just verify on the running nodes if I have stopped some.

Reply:
True the isRunning() doesn't need to exist, I just added both the getter and setters for both fields for consistency.

Reply:
This approach with NodeInfo and that with the `verifyRunningNode` are inconsistent with the indirection used elsewhere within Cluster and AT framework.

Let me think about it, there'll be a more consistent way to this goal.

Reply:
done